pkg/compactor/compactor.go

```diff
@@ -792,10 +792,20 @@ func (c *Compactor) compactUserWithRetries(ctx context.Context, userID string) e
 		if lastErr == nil {
 			return nil
 		}
+		if ctx.Err() != nil {
+			level.Warn(util_log.WithContext(ctx, c.logger)).Log("msg", "received context error during compaction", "org_id", "err", ctx.Err())
```

Comment:
Is this log even necessary? context error should only happen when compactor instance is shutting down. For me it is fine to just return the error